kolibri/core/assets/src/views/AttemptLogList.vue

```diff
@@ -63,7 +63,7 @@
         v-for="(section, index) in sections"
         :id="`section-questions-${index}`"
         :key="`section-questions-${index}`"
-        :title="displaySectionTitle(section, index) || ''"
+        :title="displaySectionTitle(section, index) || sectionLabel$({ sectionNumber: index + 1 })"
```

Comment:
Hrm, is this happening when section.title is undefined rather than being an empty string? Perhaps we could just tweak the implementation here: https://github.com/learningequality/kolibri/blob/3d12daee57a461dc157b3517ce299f1af074a169/packages/kolibri-common/strings/enhancedQuizManagementStrings.js#L224 to do a looser check, that `section` is defined, `section.title` is defined, and `section.title is not empty`?

So could maybe update it to:

```
export function displaySectionTitle(section, index) {
  return section?.section_title
    ? sectionLabel$({ sectionNumber: index + 1 })
    : section.section_title;
}
```

Reply:
Yes, this was happening because `displaySectionTitle()` is checking for `section.section_title`, but it seems that Attemptlogs for practice exercises have `section.title` instead. I did try this way, but had some issues with it not working when checking the attempt logs for exams.

```
export function displaySectionTitle(section, index) {
  const title = section.section_title || section.title;
  return title === ' '
    ? sectionLabel$({ sectionNumber: index + 1 })
    : title;
}
```

I could instead try
```
const title = section.section_title ? section.section_title : section.title;
```

But it felt like a lot more to add to the function at the time!

Reply:
Hrm... I suspect that the discrepancy between section_title and title is the root issue here. So, maybe we should trace back where that is coming from and just rectify that instead?

Reply:
i.e. it should always be `section_title` for the title for a section, and not `title`.

Reply:
Hmm, I may have found where it's coming from. This [annotateSections() util function](https://github.com/learningequality/kolibri/blob/3d12daee57a461dc157b3517ce299f1af074a169/kolibri/core/assets/src/exams/utils.js#L258).

Reply:
That looks rather like a bug, and one that I caused, looking at the git blame, so I'd be most obliged if you could fix it!

Reply:
It's fixed, now all sections annotated with `annotateSections()` have the field `section_title`
Before:
![AnnotatedSectionBefore](https://github.com/user-attachments/assets/5c63a9e8-14b2-4701-a644-5ca9c0f39582)
After:
![AnnotatedSectionAfter](https://github.com/user-attachments/assets/bd9baa51-aa95-4dc4-a48c-7300b66dacd8)

Reply:
Thank you!